python/ray/serve/tests/test_cli.py

```diff
@@ -478,5 +478,41 @@ def test_idempotence_after_controller_death(ray_start_stop, use_command: bool):
     ray.shutdown()
 
 
+@pytest.mark.skipif(sys.platform == "win32", reason="File path incorrect on Windows.")
+@pytest.mark.parametrize("config_update", [True, False])
+def test_deploy_config_update(ray_start_stop, config_update: bool):
+    config_file_name = os.path.join(
+        os.path.dirname(__file__), "test_config_files", "pid.yaml"
+    )
+    subprocess.check_output(["serve", "deploy", config_file_name])
+
+    ray.init(address="auto", namespace=SERVE_NAMESPACE)
+    serve.start(detached=True)
+
+    def num_healthy_deployments():
+        status_response = subprocess.check_output(["serve", "status"])
+        deployment_status = yaml.safe_load(status_response)["deployment_statuses"]
+        return len([x for x in deployment_status if x["status"] == "HEALTHY"])
+
+    wait_for_condition(lambda: num_healthy_deployments() == 1, timeout=5)
+    pid1 = requests.get("http://localhost:8000/f").text
+
+    config_file_name_redeploy = os.path.join(
+        os.path.dirname(__file__),
+        "test_config_files",
+        "pid_more_replicas.yaml" if config_update else "pid_code_change.yaml",
+    )
+    subprocess.check_output(["serve", "deploy", config_file_name_redeploy])
+
+    wait_for_condition(lambda: num_healthy_deployments() == 1, timeout=5)
+    pids = []
+    for _ in range(4):
```

Comment:
Could we add a comment above this for-loop mentioning that this shouldn't be flaky as long as Serve implements round-robin routing for its replicas (if that routing ever changes, this test could become mysteriously flaky)?